bakery/src/tasks/link-rex.js

```diff
@@ -0,0 +1,75 @@
+const dedent = require('dedent')
+
+const { constructImageSource } = require('../task-util/task-util')
+
+const task = (taskArgs) => {
+  const { inputSource, contentSource } = taskArgs
+  const imageDefault = {
+    name: 'openstax/cops-bakery-scripts',
+    tag: 'trunk'
+  }
+  const imageOverrides = taskArgs != null && taskArgs.image != null ? taskArgs.image : {}
+  const imageSource = constructImageSource({ ...imageDefault, ...imageOverrides })
+
+  return {
+    task: 'link rex',
+    config: {
+      platform: 'linux',
+      image_resource: {
+        type: 'docker-image',
+        source: imageSource
+      },
+      inputs: [
+        { name: 'book' },
+        { name: `${inputSource}` }
+      ],
+      outputs: [
+        { name: 'rex-linked' },
+        { name: 'common-log' }
+      ],
+      params: {
+        CONTENT_SOURCE: `${contentSource}`
+      },
+      run: {
+        user: 'root',
+        path: '/bin/bash',
+        args: [
+          '-cxe',
+          dedent`
+          exec > >(tee common-log/log >&2) 2>&1
+          case $CONTENT_SOURCE in
+            archive)
+              collection_id="$(cat book/collection_id)"
+              book_dir="${inputSource}/$collection_id"
+              xhtmlfiles_path="$book_dir/collection.mathified.xhtml"
+              abl_file="$book_dir/approved-book-list.json"
+              target_dir="rex-linked/$collection_id/"
+              mkdir "$target_dir"
+              filename="collection.rex-linked.xhtml"
+              book_slugs_file="/tmp/book-slugs.json"
+              cat $abl_file | jq ".approved_books|map(.books)|flatten" > "$book_slugs_file"
+              ;;
+            git)
+              xhtmlfiles_path="${inputSource}/*.mathified.xhtml"
+              target_dir="rex-linked"
+              filename="$(cat book/slug).rex-linked.xhtml"
+              book_slugs_file="idontexistforGit"
```

Comment:
Minor: If you want to avoid some of the checking for whether this file exists later, you could write an empty JSON like `echo "{}" > "$book_slugs_file"` so `book_slugs_by_uuid` ends up as an empty dict in the script (it would require slightly adjusting the fallback logic in `update_doc_links`, though). 🤷‍♂️